x-pack/solutions/security/plugins/security_solution/public/onboarding/components/hooks/use_url_detail.test.ts

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { renderHook, act } from '@testing-library/react';
+import { useUrlDetail, useSyncUrlDetails, getCardIdFromHash } from './use_url_detail';
+
+// --- Mocks for dependencies ---
+jest.mock('@kbn/security-solution-navigation', () => ({
```

Comment:
let's add missing `...requireActual()` calls in all these imports to make sure we don't chase hard to debug test fails whenever we use other overriden imports in the source